manifests/params.pp

```diff
@@ -52,14 +54,121 @@
   # no client certs should be trusted for auth by default.
   $ssl_certs_dir          = undef
 
-  if ($::operatingsystem == 'Ubuntu' and $::lsbdistrelease == '10.04') or ($::operatingsystem == 'SLES') {
+  if ($::apache::globals::scl) {
+    $_scl_httpd = $::apache::globals::scl_httpd
+    $_scl_php_version = $::apache::globals::scl_php_version
+    $verify_command = "/opt/rh/${_scl_httpd}/root/usr/sbin/apachectl -t"
+  } elsif ($::operatingsystem == 'Ubuntu' and $::lsbdistrelease == '10.04') or ($::operatingsystem == 'SLES') {
     $verify_command = '/usr/sbin/apache2ctl -t'
   } elsif $::operatingsystem == 'FreeBSD' {
     $verify_command = '/usr/local/sbin/apachectl -t'
   } else {
     $verify_command = '/usr/sbin/apachectl -t'
   }
-  if $::osfamily == 'RedHat' or $::operatingsystem =~ /^[Aa]mazon$/ {
+  if ($::apache::globals::scl) {
+    $user                 = 'apache'
+    $group                = 'apache'
+    $root_group           = 'root'
+    $apache_name          = "${_scl_httpd}-httpd"
```

Comment:
I'd advise introducing helpers (pseudo-code):
```puppet
if $scl {
  $_scl_httpd_prefix = "${_scl_httpd}-"
  $httpd_root = "/opt/rh/${_scl_httpd}/root"
} else {
  $_scl_httpd_prefix = ''
  $httpd_root = ''
}

$apache_name = "${_scl_httpd_prefix}httpd"
$httpd_dir = "${httpd_root}/etc/httpd"
```

Now you can use that within the RH block without any conditionals.